Join button

jsbsmp opened the pull request:
### Checklist
- [x] I've run `mvn test` from the root directory to see all new and existing tests pass
- [x] I've followed code style and run and ensured the code style is valid
- [x] I've created new tests if necessary.

### Motivation and Context
https://github.com/ctco-dev/carpooling-2018/issues/23

### Description
Added functionality to the button "Join" (from Main page) for reducing trip places number in active trips table by click.
If there are no available seats, the button "Join" is unavailable.
Created tests for new methods.

olegvm commented on src/main/java/lv/ctco/javaschool/app/control/TripStore.java:
```diff
@@ -30,4 +31,21 @@
+                .findFirst();
+    }
+
+    public void setTrip(int places, String id) {
```
The method name should match its purpose, e.g. `setTripPlaces()`.

olegvm commented on src/main/webapp/javascript/main.js:
```diff
@@ -28,4 +27,27 @@ function logout() {
+    var tripId = $(button).closest('tr').find('.table_id').text();
+    var id = $(button).closest('td').parent().index();
+    var x = document.getElementById("trips").rows[id + 1].cells;
+    x[4].innerHTML = places;
```
The variable name `x` does not explain its purpose. Please use more meaningful names.
E.g. `places` and `tripId` are OK names, while `id` does not make it clear whose id it is, and `x` is an unknown.

olegvm commented on src/main/webapp/javascript/main.js:
```diff
@@ -28,4 +27,27 @@ function logout() {
+    var data = {};
+    var places = $(button).closest('tr').find('.table_places').text() - 1;
+    var tripId = $(button).closest('tr').find('.table_id').text();
+    var id = $(button).closest('td').parent().index();
```
This code will easily break if something changes in DOM structure. Can you pass these variables as parameters to the `join()` method?`

jsbsmp commented:
- [x] renamed setTrip() method of the TripStore class to setTripPlaces()
- [x] renamed variable x to more meaningful name "rowCells"
- [x] converted variables tripId, rowId and places to join() method parameters

olegvm commented on src/main/webapp/main.jsp:
```diff
@@ -35,13 +36,16 @@
             <td>
-                <button>Join</button>
+                <button id="join-button" type="button"
+                        onclick="join(this, $(this).closest('tr').find('.table_id').text(), $(this).closest('td').parent().index(), $(this).closest('tr').find('.table_places').text())">Join
```
As per discussion, plain IDs won't work here because the markup is repeated for each row acquired from DB.
